third_party/move/move-vm/types/src/values/serialization_tests.rs

```diff
@@ -261,3 +261,30 @@ fn test_serialized_size() {
             .serialized_size(&value, &layout));
     }
 }
+
+#[test]
+fn signer_round_trip_vm_value() {
```

Comment:
Let's add more tests, also separating them into test cases helps readability. We should test different combinations:
1. Legacy signer turned on/off. And for each option:
2. Value <--> Value roundtrip
3. MoveValue to MoveValue roundtrip (you did it in `move-core-types`)
4. MoveValue <--> Value